### PR TITLE
feat: added rest of GTFS components

### LINKED ISSUE #1506: Add the rest of the GTFS Components in the Overview section
Now that we have reached a good level of consensus with the GTFS community on the list of existing GTFS Components, we want to complete their addition to this validator. 

[This spreadsheet](https://docs.google.com/spreadsheets/d/1iqvzJV_YWmFyYGtpbO2dqGMbf4XEjvar3rt9SxHU-xY/edit#gid=0) contains the list of all GTFS Components, and the conditions under which we can them "present" in the validator. 
The task is to add all components that have not been added yet. 

Labelling this issue as "epic" because we might want to break it down into smaller increments.

### LINKED ISSUE #1513: Modify the "Fares v2" component to "Fare Products"
### Describe the problem

Now that we will are adding more components as part of GTFS fares v2 (fare Media, time-based Fares, etc), we need to rename the "GTFS fares v2" component to "Fare products". 
See the [spreadsheet](https://docs.google.com/spreadsheets/d/1iqvzJV_YWmFyYGtpbO2dqGMbf4XEjvar3rt9SxHU-xY/edit#gid=0) for reference. 

### Proposed solution

Rename "Fares v2" to "Fare Products" in the list of GTFS Components.

### Alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cka-y**
**Summary:**

[This spreadsheet](https://docs.google.com/spreadsheets/d/1iqvzJV_YWmFyYGtpbO2dqGMbf4XEjvar3rt9SxHU-xY/edit#gid=0) contains the list of all GTFS Components, and the conditions under which we can them "present" in the validator.
All adopted components are added.

Closes #1506 
Closes #1513 

**Expected behavior:** 

image

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java [hunk 1]
```diff
@@ -85,23 +86,175 @@ private <E extends GtfsEntity> int loadUniqueCount(
   }
 
   private void loadSpecFeatures(GtfsFeedContainer feedContainer) {
+    loadSpecFeaturesBasedOnFilePresence(feedContainer);
+    loadSpecFeaturesBasedOnFieldPresence(feedContainer);
+  }
+
+  private void loadSpecFeaturesBasedOnFilePresence(GtfsFeedContainer feedContainer) {
+    List<Pair<String, String>> componentList =
```
Comment (davidgamez) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java:94]: This variable can be converted to a constant.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java [hunk 2]
```diff
@@ -85,23 +86,175 @@ private <E extends GtfsEntity> int loadUniqueCount(
...
+            new Pair<>("Attributions", GtfsAttribution.FILENAME),
+            new Pair<>("Translations", GtfsTranslation.FILENAME),
+            new Pair<>("Fare Media", GtfsFareMedia.FILENAME),
+            new Pair<>("Zone-Based Fares", GtfsStopArea.FILENAME),
+            new Pair<>("Transfers", GtfsTransfer.FILENAME));
+
+    for (Pair<String, String> entry : componentList) {
+      specFeatures.put(
+          entry.getKey(),
+          hasAtLeastOneRecordInFile(feedContainer, entry.getValue()) ? "Yes" : "No");
```
Comment (davidgamez) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java:112]: The _"Yes"_ and  _"No"_ values can be converted to constants.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java [hunk 3]
```diff
@@ -85,23 +86,175 @@ private <E extends GtfsEntity> int loadUniqueCount(
...
+    }
+  }
+
+  private void loadSpecFeaturesBasedOnFieldPresence(GtfsFeedContainer feedContainer) {
```
Comment (davidgamez) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java:116]: I think there is an opportunity here to reduce the number of lines for the method following the same pattern as before having a single method per component. Having shorter methods improve the readability and maintainability of the code.

Comment (cka-y) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java:116]: just to make sure i understand properly, you suggest we separate this method in multiple small methods (one per component) for improved readability?

Comment (davidgamez) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java:116]: Yes, that's the suggestion.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java (hunk 2 above)
Comment (cka-y) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java]: I updated the code and removed it completely. True & false was good enough for our use if the values.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java [hunk 4]
```diff
@@ -12,11 +13,25 @@ public class FeedMetadata {
...
 
   public ArrayList<AgencyMetadata> agencies = new ArrayList<>();
   private ImmutableSortedSet<String> filenames;
 
+  private final List<Pair<String, String>> FILE_BASED_COMPONENTS =
+      List.of(
+          new Pair<>("Pathways", GtfsPathway.FILENAME),
+          new Pair<>("Transfers", GtfsTransfer.FILENAME),
+          new Pair<>("Fares V1", GtfsFareAttribute.FILENAME),
+          new Pair<>("Fare Products", GtfsFareProduct.FILENAME),
```
Comment (davidgamez) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java:26]: 💪

**Review by davidgamez — APPROVED** [review]
💪

**Pull request merged**